udata_gouvfr/tasks.py

```diff
@@ -0,0 +1,63 @@
+import mongoengine
+import requests
+
+from collections import defaultdict
+from flask import current_app
+
+from udata.commands import success, error
+from udata.core.dataset.models import Dataset
+from udata.tasks import job
+
+from udata_gouvfr import (
+    APIGOUVFR_EXTRAS_KEY,
+    APIGOUVFR_EXPECTED_FIELDS,
+)
+
+
+def get_dataset(id_or_slug):
+    obj = Dataset.objects(slug=id_or_slug).first()
+    return obj or Dataset.objects.get(id=id_or_slug)
+
+
+def process_dataset(d_id, apis):
+    try:
+        dataset = get_dataset(d_id)
+    except (Dataset.DoesNotExist, mongoengine.errors.ValidationError):
+        return error(f'Dataset {d_id} not found')
+    dataset.extras[APIGOUVFR_EXTRAS_KEY] = apis
+    dataset.save()
+    success(f'Imported {len(apis)} API(s) for {str(dataset)}')
+
+
+@job('apigouvfr-load-apis')
+def apigouvfr_load_apis(self):
+    '''Load dataset-related APIs from api.gouv.fr'''
+    r = requests.get(current_app.config['APIGOUVFR_URL'], timeout=10)
+    r.raise_for_status()
+
+    # cleanup existing mappings
+    Dataset.objects.filter(**{
+        f'extras__{APIGOUVFR_EXTRAS_KEY}__exists': True,
+    }).update(**{
+        f'unset__extras__{APIGOUVFR_EXTRAS_KEY}': True,
+    })
```

Comment:
Is the filter/exists needed before the update/unset? I didn't do this in another plugin and it seemed fine. I'm convinced this code works and is readable, just a curiosity

Reply:
It's probably (a lot?) faster w/ the filter but it shouldn't change the outcome.

Reply:
Thanks for the feedback 🙏 https://github.com/opendatateam/udata-recommendations/pull/142/commits/d05d0359cc1298623a128c18c29eea1d2e516557